src/native/external/llvm-libunwind/src/DwarfInstructions.hpp

```diff
@@ -230,12 +230,21 @@ int DwarfInstructions<A, R>::stepWithDwarf(A &addressSpace, pint_t pc,
         p &= ~0xfULL;
         // CFA is the bottom of the current stack frame.
         for (; p < cfa; p += 16) {
+#ifdef __GNUC__
```

Comment:
clang defines `__GNUC__` for compat, so this is `#if 1` for all practical purposes.

Reply:
hmm, that is not very useful. Is there a way to detect GCC, or do we ever compile this file with GCC?

Reply:
GCC does not document `mte` as something it would accept.

Reply:
We can just put `#if 0` around this whole block for now. We do not need support for mtag in our copy of libunwind.

Reply:
Although it might be that both clang and gcc support mte:

[Trying use it in godbolt](https://godbolt.org/#z:OYLghAFBqd5QCxAYwPYBMCmBRdBLAF1QCcAaPECAMzwBtMA7AQwFtMQByARg9KtQYEAysib0QXACx8BBAKoBnTAAUAHpwAMvAFYTStJg1DIApACYAQuYukl9ZATwDKjdAGFUtAK4sGe1wAyeAyYAHI%2BAEaYxBLSAA6oCoRODB7evnoJSY4CQSHhLFExUraY9jkMQgRMxARpPn5cpeUpVTUEeWGR0bG21bX1GU0K/R3BXYU9UgCUtqhexMjsHCYaAIKra8EEANTAmAQA%2BgBuYl6YENs7qtM7myYA7FbrO6872wBUO3F3AMwAIjsNCZfs81m8IW9DocmAoWNCdtDjp4mI56NCIOYzAA6GrIBA7GosY4ADmxAFYALRMEzktwMLGbSHMlmsyFY3GLBCHTCqAiMJICHYsfm0%2BmMl5sqXSrEjYB3Mzk2kWZQEYi0/6kO7kqyK5Wq9Xk/4asUMsxmJnSq0skCW63214gbUqtUahVmdXmiBxaZ2h3Wp1YtgsEgATyxvtBluIBwWDB2lK4ILBj2N6w4s1onHJvD8HC0pFQnDc1msOwU80WmAVvx4pAImgzswA1iBfgBObHt8mSSRmSQANi4Ggew7MTSzHEkvBYEg0GlIeYLRY4vAUIAXDfzGdIcFgMEQKFQLDidGi5EoaBPZ5iRIHkmAaq8DGbpCwxzwSwAanhMAB3AB5OJGE4OsaFofliHXCAIkbUgImCGpQ1A3gEOYYhQwAiJtEwBwUNIK82EEACGFoZDtzfTAWEMYBxAo/AYwcPBjkwdcKN5XCvH5fDtjKfDaDwCJiCQjwsHwtU8FnbgdyoAxgAUH9/yAkDpJkQQRDEdgSn4QRFBUdQKN0Mx9BolBS0sfRBPXSBZlQOIKjYtcylwioXAYdxPAafx3M6AoikyRJkgEQZGlILIgoYXzumKZoXNaUYQr0Ow4oENpaiiyYYpGdpEuGUYMv8rhZgrBYlgkTNs1zOCVx2NYACUAFl7z2ZBkB2J8Xx2CBcEIEgayK3gty0WYICQeYCDiLiLwgTB8CIYg8HQPQdOEURxG02Q9LUOCjNIP9hLiFCKo4HNF2qzgAK4ybdlQKhasa5rgFa9riGfZsuo8a96GIGszGmQbG2mFsQDMMlflHSQuAHDQB3bX5yQeSRfmkSdp1IWdJA0bENC4dtQfvDRfg0ElyRJYyl14Fc1w3etAd3A9RqPL7zwoCAr1Pb6QCJUlKSoLxaFoSlkAMIxKQ61930/TBFMA4C8zAuhIOg2CKLQpD8LVjCsJwvDVMIxgCBIsi4KwaijDousGLili2ILDjkC45Y614yc6wEoSRIwJ3BoWqSeFmWSmHkmXlPl3gVo09bpBWraDILXQJ1M0xLGsSyImsiBbPslI2ITADfgTKhTzapyWmcGb3Ny0hAnGPypjCwKKir8KKgK%2BvkqY1KEs8oZYs7yp8tr6Kku79JQuy9Kh8y8q5lKrTjtOinC04QliGJElqR2YWaK68WtTEWhV/xQgXIWMQdioTBUQWVjbm6ua%2BvMWt/tp7cgf0Tg0dnclO3bEkRy4PDdssMpBcAeGdCiVNbA0yGk2UgrZobY0kOSBG5Jfgkjxhgh47YP4cF%2BFVSBnAAZv2OmYAhy4iGv2GrMFiUEUggEkEAA%3D%3D)

Reply:
> Is there a way to detect GCC

We can instead use `#ifdef __llvm__` (if it is really needed)

Please reference the commit modifying external source in `src/native/external/llvm-libunwind-version.txt`.

Reply:
The problem is with older versions of llvm that do not understand this extension at all.

Reply:
Maybe just replace it with `assert(false)` until we have to support ‘mte’?  
We’d have to switch to compiler that supports it by then.

Reply:
We are setting `stage2` to false: https://github.com/dotnet/runtime/pull/85179/files#diff-342f897208801d38daad0d3b917fae03bdcc260357ebfc132231be828977dae0, so we can just skip this branch as @jkotas has done in https://github.com/dotnet/runtime/pull/85431.